bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/PathTest.java

```diff
@@ -318,6 +318,18 @@ public void testConstructors() {
 		assertEquals("3.0.win", "D:/foo/abc.txt", IPath.forWindows("/D:/foo/abc.txt").toString());
 		// fullPath = new java.io.File("D:/").toURL().getPath()
 		assertEquals("3.1.win", "D:/", IPath.forWindows("/D:/").toString());
+
+		final String verylongpath = "C:/dev/verylongpath/00112233445566778899aabbccddeeff/00112233445566778899aabbccddeeff/00112233445566778899aabbccddeeff/"
+				+ "00112233445566778899aabbccddeeff/00112233445566778899aabbccddeeff/00112233445566778899aabbccddeeff/root/lib/gcc/arm-none-eabi/7.3.1/include-fixed";
+		anyPath = IPath.forWindows(verylongpath);
+		assertEquals("3.2.1.win", "C:", anyPath.getDevice());
+		assertEquals("3.2.2.win", "dev", anyPath.segment(0));
+		assertEquals("3.2.3.win", verylongpath, anyPath.toString());
+
+		anyPath = IPath.forWindows("//?/" + verylongpath);
```

Comment:
On Windows backslashes are used as separator by default and as the linked guide mentions `\\?\` it should be tested literally with that value (even if the Path constructor replaces `\` by `/`).
```suggestion
		anyPath = IPath.forWindows("\\\\?\\" + verylongpath);
```

Reply:
I added a test for both variants, is this okay for you?